backend/src/org/commcare/resources/model/ResourceTable.java

```diff
@@ -126,17 +126,17 @@ public void removeResource(Resource resource) {
         storage.remove(resource);
     }
 
-    public void addResource(Resource resource, ResourceInstaller initializer, String parentId, int status) throws StorageFullException {
+    public void addResource(Resource resource, ResourceInstaller initializer, String parentId, int status) {
         resource.setInstaller(initializer);
         resource.setParentId(parentId);
         addResource(resource, status);
     }
 
-    public void addResource(Resource resource, ResourceInstaller initializer, String parentId) throws StorageFullException {
+    public void addResource(Resource resource, ResourceInstaller initializer, String parentId) {
         addResource(resource, initializer, parentId, Resource.RESOURCE_STATUS_UNINITIALIZED);
     }
 
-    public void addResource(Resource resource, int status) throws StorageFullException {
```

Comment:
Sneaking this change into this PR:
addResource never throws the StorageFullException runtime exception, since it explicitly swallows it. Hence we should take it out of the method signature completely.